### DISCUSSION TITLE
single line vs structured format in .json example

### DISCUSSION THREAD

**Discussion by jltang-kp** [CMSgov/price-transparency-guide#187] [Q&A] [answered]
We have been using MS azure to output our machine readable files and we have been running into an issue where azure will only output on a single line and not like the sample where it is structured. 

I don't believe there should be a difference if the machine readable file is consumed by an appropriate reader though. However would like to get some others opinion if the files have to be in a structured format.

**Comment by ghost** [comment, marked as answer]
My .02 if it is a properly formatted as in the case of a JSON file, it should be okay even it it was not "beutified" as the samples are for us to read while the files we generate are to be machine readable.

**Reply by vijayanandramamoorthy** [reply]
I agree. Actually, I believe it is desirable to post the files with out "beautification", as formatting the file will increase the file size by a few gigabytes.

**Reply by jltang-kp** [reply]
Thanks, appreciate the inputs. I'm glad others are under the same thinking that the "beautification" is cosmetic only.

**Reply by shaselton-usds** [reply]
100% agree here -- compress these files as much as possible (e.g. eliminate white-space). After the file is downloaded, "beautification" can be applied for those that wish to look directly at the content.

**Reply by saikiran4u** [reply]
But the CMS Validator does not seem to successfully parse/Accept single line JSON. Any suggestions on how to get around this problem.JSON is created by an ETL tool(SSIS).

**Reply by jltang-kp** [reply]
@saikiran4u what sort of error messages are you seeing? We have run a single line JSON through the validator and error messages look different than that of having incorrect values
